rkt/pods.go

```diff
@@ -749,6 +754,47 @@ func (p *pod) getState() string {
 	return state
 }
 
+func (p *pod) getModTime(path string) (time.Time, error) {
+	f, err := p.openFile(path, syscall.O_RDONLY)
+	if err != nil {
+		return time.Time{}, err
+	}
+
+	fi, err := f.Stat()
+	if err != nil {
+		return time.Time{}, err
+	}
+
+	return fi.ModTime(), nil
+}
+
+// getCreationTime returns the time when the pod was created.
+// This happens at prepare time.
+func (p *pod) getCreationTime() (time.Time, error) {
+	if p.isPrepared || p.isRunning() || p.afterRun() {
+		return p.getModTime("pod")
+	}
+	return time.Time{}, nil
+}
+
+// getStartTime returns the time when the pod was started.
+func (p *pod) getStartTime() (time.Time, error) {
+	var (
+		t   time.Time
+		err error
+	)
+	if p.isRunning() || p.afterRun() {
+		t, err = p.getModTime("ppid")
```

Comment:
At the moment, stage1 implementations have the choice to write either the "ppid" or the "pid. Then, [stage0's getContainerPID1()](https://github.com/coreos/rkt/blob/master/rkt/pods.go#L826) is able to read either file: it tries "pid" and if it does not work, fall back on "ppid".

The two files are documented in the [Stage 1 ACI implementor's guide](https://github.com/coreos/rkt/blob/master/Documentation/devel/stage1-implementors-guide.md#rkt-run--coreoscomrktstage1run).

To be consistent, getStartTime() should check "pid" too, although it is not very useful since it is not written in the current implementation...